File: c++/0001_TwoSum.cpp
// Problem 1: Two Sum
// https://leetcode.com/problems/two-sum/

#include <vector>
using namespace std;

class Solution 
{
public:
    vector<int> twoSum(vector<int>& nums, int target)
    {
        // 宣告回傳的型態
        vector<int> result;
        // 使用雙層迴圈, 第二層的起點為 i+1 避免重複
        for (int i = 0; i < nums.size(); i++){
            for (int j = i + 1; j < nums.size(); j++){
                // 相加等於 target 即找到, 離開迴圈
                if (nums[i] + nums[j] == target){
                    result.push_back(i);
                    result.push_back(j);
                    break;
                } 
            }
        }    
    return result;
    }
};

File: python/0001_two_sum.py
# Problem 1: Two Sum
# https://leetcode.com/problems/two-sum/

from typing import List


class Solution:
    def two_sum(self, nums: List[int], target: int) -> List[int]:

        # 使用 enumerate 取得當前 index 和 數值
        for i, num in enumerate(nums):
            # 如果 希望的值(目標減去當前數值) 在這個list中 且 不是同一個位置的值(index不相同)
            # 就視為找到答案, 直接return
            if target - num in nums and i != nums.index(target - num):
                return i, nums.index(target - num)
